src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/RSACng.ImportExport.cs

```diff
@@ -38,10 +38,13 @@ public override void ImportParameters(RSAParameters parameters)
             unsafe
             {
                 if (parameters.Exponent == null || parameters.Modulus == null)
-                    throw new ArgumentException(SR.Cryptography_InvalidRsaParameters);
+                    throw new CryptographicException(SR.Cryptography_InvalidRsaParameters);
 
                 bool includePrivate = parameters.P != null && parameters.Q != null;
 
+                if (includePrivate && parameters.DP == null)
```

Comment:
Why just DP? Why not also D, DQ, and InverseQ?

The state is that D, P, Q, DP, DQ, and InverseQ must either all be null, or all be not-null.